reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -78,38 +89,229 @@
+                            .equals(
+                           currentGav.getGA().getArtifactId())
+                            
+                            &&
+                           
+                           x.getGa().getGroupId()
+                            .equals(
+                           currentGav.getGA().getGroupId());
+                })
+                .collect(Collectors.toSet());
+                
+        List<String> whiteVersionsForCurrentGav = whiteArtifactsForCurrentGav.stream()
+                .map(x -> x.getVersion())
+                .collect(Collectors.toList());
+        
+        List<String> availableVersionsFromWhitelist = versionFinder.getBuiltVersionsFor(currentGav, whiteVersionsForCurrentGav);
+        return availableVersionsFromWhitelist;
+    }
+
+    private Set<Long> getProductVersionIds(Set<ProductVersion> relevantProductVersions) {
+        return relevantProductVersions.stream()
+                .map(x -> x.getId())
+                .collect(Collectors.toSet());
+    }
+
+    private Set<ProductVersion> getProductVersions(Set<String> productNames, Set<Long> productVersionIds) {
+        Set<ProductVersion> productVersions = new HashSet<>();
+        StringBuilder errorMsg = new StringBuilder();
+        
+        if(productNames != null && !productNames.isEmpty()){
+            List<org.jboss.da.listings.api.model.Product> productsByName = productDao.findAllWithNames(new ArrayList<>(productNames));
+            if(productsByName!= null && (productNames.size() == productsByName.size())){
+                for(String productName : productNames){
+                    List<ProductVersion> prodVers = productVersionService.getAllForProduct(productName); 
+                    productVersions.addAll(prodVers);
+                }
+            } else {
+                // Error
+                Set<String> inexistingProductNames = new HashSet<>(productNames);
+                productsByName.stream().forEach(x -> inexistingProductNames.remove(x.getName()));
+                errorMsg.append("Product names do not exist: ");
+                appendErrors(errorMsg, inexistingProductNames); 
+            }
+
+        }
+        
+        if(productVersionIds != null && !productVersionIds.isEmpty()){
+            List<ProductVersion> prodVersionsById = productVersionDao.findAllWithIds(new ArrayList<>(productVersionIds));
+            if(prodVersionsById != null && (productVersionIds.size() == prodVersionsById.size())){
+                productVersions.addAll(prodVersionsById);
+            } else {
+                // Error
+                Set<Long> inexistingProductVersionIds = new HashSet<>(productVersionIds);
+                prodVersionsById.stream().forEach(x -> inexistingProductVersionIds.remove(x.getId()));
+                errorMsg.append("Product Versions do not exist: ");
+                appendErrors(errorMsg, inexistingProductVersionIds); 
+            }
+        }
```

Comment:
What if I enter several productVersion ids and one of them is invalid? I would say we should report an error, but here we ignore it. The same for product name.